Tribler/Core/Upgrade/pickle_converter.py

```diff
@@ -0,0 +1,182 @@
+import io
+import os
+import glob
+import json
+import pickle
+import cPickle
+import StringIO
+
+from ConfigParser import RawConfigParser
+
+from Tribler.Core.SessionConfig import SessionStartupConfig, SessionConfigInterface
+from Tribler.Core.DownloadConfig import DefaultDownloadStartupConfig, get_default_dscfg_filename
+from Tribler.Core.simpledefs import PERSISTENTSTATE_CURRENTVERSION, STATEDIR_GUICONFIG
+
+
+class PickleConverter(object):
+    """
+    This class is responsible for converting old .pickle files used for configuration files to a newer .state format.
+    """
+
+    def __init__(self, session):
+        self.session = session
+
+    def convert(self):
+        """
+        Calling this method will convert all configuration files to the newer .state format.
+        """
+        self.convert_session_config()
+        self.convert_main_config()
+        self.convert_default_download_config()
+        self.convert_download_checkpoints()
+
+    def convert_session_config(self):
+        """
+        Convert the sessionconfig.pickle file to libtribler.conf. Do nothing if we do not have a pickle file.
+        """
+        old_filename = os.path.join(self.session.get_state_dir(), 'sessconfig.pickle')
+        new_filename = SessionConfigInterface.get_default_config_filename(self.session.get_state_dir())
+
+        if not os.path.exists(old_filename):
```

Comment:
Do we know what was the last version of Tribler using pickled sessions? I'm pretty sure it's been years. We probably could just remove this code.